### DISCUSSION TITLE
Service integration

### DISCUSSION THREAD

**Discussion by jesperpedersen** [Hyperfoil/Horreum#376] [Ideas]
We currently depend on `docker-compose` or `podman-compose` to setup the service dependencies that Horreum relies on.

Using a compose script of course makes it easier for people to get the entire platform running, but it also highlights some areas where improvements could be made.

F.ex. https://github.com/Hyperfoil/Horreum/blob/master/doc/DEVELOPMENT.md#using-existing-data highlights that changes are needed to files if you need to use existing data.

What does people think about removing the dependency on the compose script completely ? We would have scripts for individual services and other scripts that would combine these scripts to start/stop each service. This would likely make it easier to test new KeyCloak releases.

I think we can keep the `docker` or `podman` requirement to run the external services for us.

Feel feel to share your ideas on this subject !

**Comment by johnaohara** [comment]
The infra scripts /podman/docker compose scripts only work on linux, and do not work on any other platform. I have been thinking about migrating the infra to quarkus dev services, so that the only command required to start up the dev environment is 'mvn quarkus:dev' or 'quarkus dev' is a user has qarkus dev cli installed. The dev services could be congurable such that if a backup is used to stand up the dev env. then no changes are required

**Reply by jesperpedersen** [reply]
If we can get to a point where we can remove `infra/` completely that would be ideal. But, maybe we need different development "profiles" where an existing service is used with minimal changes to the configuration.

**Comment by johnaohara** [comment]
I don't have any code changes to share atm, I'll share next week when I am back in front of my desktop machine
